### PR TITLE
Emit Job creation failed event

### LINKED ISSUE #447: Emit event when Job creation fails
**What would you like to be added**:
Emit event when Job creation fails

**Why is this needed**:
To propagate the Job creation failure up to be more visible to the user, so they don't need to dig through the logs to know that their Job creation failed.

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Fixes #447 

I opted to only emit 1 event if any Job creation fails, rather than emit a separate event for each Job failure (since there could be many).

While unconventional, I included the error message in the event message since the intent is to allow the user to quickly see why Jobs aren't being created without digging through logs.

### CODE REVIEW

File: pkg/controllers/jobset_controller.go [hunk 1]
```diff
@@ -475,6 +475,9 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 	}
 	allErrs := errors.Join(finalErrs...)
 	if allErrs != nil {
+		// Emit event to propagate the Job creation failures up to be more visible to the user.
+		// TODO(#422): Investigate ways to validate Job templates at JobSet validation time.
+		r.Record.Eventf(js, corev1.EventTypeWarning, "JobCreationFailed", "Job creation(s) failed with error: %s", allErrs)
```
Comment (kannon92) [inline pkg/controllers/jobset_controller.go:480]: Could you add a bit more information to this event? If we had multiple jobs, which job creation failed? I guess validation would be in logs or kubetctl somewhere so they could dig into this.

Comment (danielvegamyhre) [reply pkg/controllers/jobset_controller.go:480]: I think the `allErrs` variable should have these details

Comment (kannon92) [reply pkg/controllers/jobset_controller.go:480]: Are we sure that events actually shows the errors in all its entirity?

Comment (danielvegamyhre) [reply pkg/controllers/jobset_controller.go]: To be safe, I wrapped the `err` returned by `r.Create(...)` in a custom error which includes the Job name at the beginning of the error message, let me know what you think of this

**Review by kannon92 — COMMENTED** [review]
/lgtm
/hold

maybe @ahg-g wants to weigh in.

**Pull request merged**